### ISSUE TITLE
BC break starting with 4.0.1

### ISSUE THREAD

**Issue by boesing** [phpredis/phpredis#1401]
### Expected behaviour
`exists` returns boolean

### Actual behaviour
`exists` returns integer, depending on how many hits the key(s) has/have

### I'm seeing this behaviour on
- OS: osx
- Redis: 4.0.11
- PHP: 7.2
- phpredis: 4.1.1

### Steps to reproduce, backtrace or example script
```php
$instance = new Redis();
$instance->connect('localhost', 6379);
$instance->set('foo', 'bar');
$instance->exists('foo'); // returns 1
```

### I've checked
- [X] There is no similar issue from other users
- [X] Issue isn't fixed in `develop` branch

**Comment by boesing** [comment]
I would suggest to return `boolean` if one key was provided.
If multiple keys are provided, return integer.

Maybe change the return value to integer only in the next major version instead?

**Comment by courtney-miles** [comment]
I was struck by this also.  Now I found that Redis itself had changed the behaviour of `exists` I think it is perhaps acceptable that returns an int and that it does this since version 4.

It would be good to get confirmation that this is a deliberate change and won't be reverted.

**Comment by michael-grunder** [comment]
Hi,

This change to the Redis API has actually caused a lot of problems for phpredis users over the years.  I introduced the change from `boolean` to `long` after the Redis API changed in a phpredis 3.x release but didn't account for the fact that it was a breaking change, so it was reverted.

We then officially made the breaking change in 4.0.0:
https://pecl.php.net/package/redis/4.0.0
https://github.com/phpredis/phpredis/blob/4.0.0/redis.c#L1168

This is made more confusing by the fact that we introduced a regression only in 4.0.1 (simple merge oversight) which reverted the code back to boolean.

TL;DR:  3.x will return true/false, 4.x will return long* (except 4.0.1 specifically).

It's always interesting which issues cause the most pain, and this is definitely in the top 3 for phpredis.  :smile:
